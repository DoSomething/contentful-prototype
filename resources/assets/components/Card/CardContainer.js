import { connect } from 'react-redux';
import { ModalListener } from '../Modal';
import { closeModal } from '../../actions/modal';
import Card from './Card';

const actionCreators = {
  closeModal,
};

export default connect(null, actionCreators)(ModalListener(Card));
