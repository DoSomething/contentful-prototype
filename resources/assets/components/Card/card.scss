@import '../../scss/next-toolbox.scss';

.card {
  background-color: $white;
  overflow: hidden;
}

.card__title {
  color: $black;
  background-color: $primary-color;
  padding: $half-spacing;

  h1 {
    font-family: $primary-font-family;
    font-size: $font-regular;
    text-transform: uppercase;
  }

  a {
    color: $black;

    &:hover {
      color: rgba($black, 0.75);  // @TODO should be a variable
      text-decoration: none;
    }
  }
}
