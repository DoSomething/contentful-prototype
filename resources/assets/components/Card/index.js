export CardContainer from './CardContainer';

export default from './Card';
