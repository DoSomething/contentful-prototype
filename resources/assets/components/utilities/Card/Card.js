import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const renderHeader = (title, link, onClose) => (
  <header className="text-gray-900 bg-yellow-500 p-3 rounded-t clip-padding-box">
    <h1
      data-test="card-title"
      className="inline font-source-sans text-base uppercase"
    >
      {link ? (
        <a
          className="text-gray-900 hover:opacity-75 hover:no-underline"
          href={link}
        >
          {title}
        </a>
      ) : (
        title
      )}
    </h1>
    {onClose ? (
      <button
        type="button"
        className="float-right text-xl -mt-2 hover:cursor-pointer"
        onClick={onClose}
      >
        &times;
      </button>
    ) : null}
  </header>
);

const Card = ({
  children,
  className = '',
  link = null,
  title = null,
  onClose,
}) => (
  <article
    className={classnames('card bg-white w-full clip-padding-box', className)}
  >
    {title ? renderHeader(title, link, onClose) : null}

    {children}
  </article>
);

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
  className: PropTypes.string,
  link: PropTypes.string,
  onClose: PropTypes.func,
  title: PropTypes.string,
};

Card.defaultProps = {
  className: null,
  link: null,
  onClose: null,
  title: null,
};

export default Card;
