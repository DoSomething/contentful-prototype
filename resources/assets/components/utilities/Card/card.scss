@import '../../../scss/next-toolbox.scss';

.card {
  background-color: $white;
  overflow: hidden;
  width: 100%;

  p + p {
    margin-top: 8px;
  }
}

.card__title {
  color: $black;
  background-color: $primary-color;
  padding: $half-spacing;

  h1 {
    font-family: $primary-font-family;
    font-size: $font-regular;
    text-transform: uppercase;
    display: inline;
  }

  a {
    color: $black;

    &:hover {
      color: rgba($black, 0.75); // @TODO should be a variable
      text-decoration: none;
    }
  }
}

.affiliate-content {
  .card__title {
    background-color: $sponsor-accent;

    h1,
    a {
      color: $white;
    }
  }
}

.card__exit {
  float: right;
  font-size: 26px;
  margin-top: -6px;

  &:hover {
    cursor: pointer;
  }

  &.-floating {
    position: absolute;
    top: $half-spacing;
    right: $half-spacing;
  }
}
