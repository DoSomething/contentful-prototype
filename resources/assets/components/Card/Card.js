import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './card.scss';

const renderHeader = (title, link, onClose) => (
  <header className="card__title">
    { link ? <h1><a href={link}>{title}</a></h1> : <h1>{title}</h1> }
    { onClose ? <button className="card__exit" onClick={onClose}>&times;</button> : null }
  </header>
);

const Card = ({ children, className = '', link = null, title = null, onClose }) => (
  <article className={classnames('card', className)}>
    { title ? renderHeader(title, link, onClose) : null }

    { children }
  </article>
);

Card.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
  className: PropTypes.string,
  link: PropTypes.string,
  onClose: PropTypes.func,
  title: PropTypes.string,
};

Card.defaultProps = {
  className: null,
  link: null,
  onClose: null,
  title: null,
};

export default Card;
