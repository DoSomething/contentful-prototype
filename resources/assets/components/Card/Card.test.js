import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Card from './index';


const component = shallow(
  <Card
    className='bordered padded rounded'
    children='Praesent commodo cursus magna, vel scelerisque nisl'
    title='Awesomest Content Ever'
  />,
);

test('it generates a card snapshot', () => {
  const tree = shallowToJson(component);

  expect(tree).toMatchSnapshot();
});

test('it can display a card as an article element', () => {
  expect(component.find('article')).toHaveLength(1);
});
