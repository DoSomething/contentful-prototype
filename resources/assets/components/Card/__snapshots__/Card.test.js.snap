// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it generates a card snapshot 1`] = `
<article
  className="card"
>
  <header
    className="card__title"
  >
    <h1>
      Awesomest Content Ever
    </h1>
  </header>
  <p>
    Praesent commodo cursus magna, vel scelerisque nisl
  </p>
</article>
`;
